First prj

dizhechko opened the pull request.

dizhechko committed:
first

dizhechko committed:
second

ivannizh commented:
Наталья, привет!

Ошибка при выполнении тестов
![image](https://github.com/user-attachments/assets/63823111-ee24-4512-a024-c90d1e791b3f)

Так же невозможно завершить задачу "Оплатить коммуналку"
![image](https://github.com/user-attachments/assets/4524f57b-ef76-46cc-8e82-edba1e8ac7dd)

dizhechko committed:
test bag fix

dizhechko committed:
Update scheduler.go

Добавлен функционал выполнения задачи по дням недели. Но тесты падали из-за параметра в tests/settings.go , так не реализован "поиск", "параметры повторения по неделям и месяцам", "авторизация". Поставила var FullNextDate = false. Нет уже возможности реализовать месяцы и авторизацию.

dizhechko commented:
Добавлен функционал выполнения задачи по дням недели. Но тесты падали из-за параметра в tests/settings.go , так не реализован "поиск", "параметры повторения по неделям и месяцам", "авторизация". Поставила var FullNextDate = false. Нет уже возможности реализовать месяцы и авторизацию.

ivannizh commented on database/database.go:
```diff
@@ -0,0 +1,188 @@
+	task := Task{}
+	err := s.db.Get(&task, "SELECT id, date, title, comment, repeat FROM scheduler WHERE id = ?", id)
+	if err != nil {
+		if err.Error() == "sql: no rows in result set" {
```
Лучше так
```suggestion
		if errors.Is(err, sql.ErrNotFound) {
```

ivannizh commented on database/database.go:
```diff
@@ -0,0 +1,188 @@
+	// Получаем ID последней записи
+	lastInsertId, err = resultDB.LastInsertId()
+	if err != nil {
+		return 0, err
```
Лучше оборачивать ошибки, чтобы быстрее понимать, где они произошли
```suggestion
		return 0, fmt.Errorf("error while grtting last id: %w", err)
```

ivannizh commented on handlers/handlers.go:
```diff
@@ -0,0 +1,377 @@
+	if err != nil {
+		log.Printf("NextDateHandler: %v %v %v %v; error: %v\n", now, date, repeat, nextDate, err)
+		w.WriteHeader(http.StatusBadRequest)
+		_, _ = w.Write([]byte(""))
```
Отлично сделано, что ошибка идет в лог, а пользователю  пустой ответ, но можно и заглушку
```suggestion
		_, _ = w.Write([]byte("Internal server error"))
```

ivannizh reviewed (approved):
Наталья, привет!

Отличная выпускная работа! Код хорошо разделен на папки и функции, все тесты проходят!

Работа принята! Желаю тебе успехов в дальнейшем изучении и применении Go!
